### PR TITLE
Fixing issue with help progress

### PR DESCRIPTION
**Pull request by powercode**
The Get-Help cmdlet never calls WriteProgress with a RecordType set to Completed, so the progress is left being rendered even after the Get-Help cmdlet has completed.

The PR is adding a write progress call with recordtype set to Completed.

It also cleans up lots of chains to `this.Context.HelpSystem` by extracting a variable.

  

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@powercode Love the change! Could you add some tests? Something like the following.

```PowerShell
Describe "Testing Get-Help Progress" {
 It "Last ProgressRecord should be Completed" {
  $j = Start-Job { Get-Help DoesNotExist }
  $j | Wait-Job
  $j.ChildJobs[0].Progress[-1].RecordType | Should -Be ([System.Management.Automation.ProgressRecordType]::Completed)
}}
```

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 [hunk 1]
```diff
@@ -96,7 +98,15 @@ Describe "Validate that get-help works for CurrentUserScope" -Tags @('CI') {
     }
 }
 
-Describe "Validate that get-help works for AllUsers Scope" -Tags @('Feature','RequireAdminOnWindows', 'RequireSudoOnUnix') {
+Describe "Testing Get-Help Progress" -Tags @('CI') {
```
Comment (iSazonov) [inline test/powershell/engine/Help/HelpSystem.Tests.ps1]: I expect that it is slow tests so we should make it 'Feature'.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 2]
```diff
@@ -334,9 +335,10 @@ protected override void ProcessRecord()
             }
             finally
             {
-                this.Context.HelpSystem.OnProgress -= new HelpSystem.HelpProgressHandler(HelpSystem_OnProgress);
+                helpSystem.OnProgress -= new HelpSystem.HelpProgressHandler(HelpSystem_OnProgress);
+                HelpSystem_OnComplete();
                 // finally clear the ScriptBlockAst -> Token[] cache
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs]: Please add new line before comment.

**Review by iSazonov — APPROVED** [review]
LGTM with one style comment.

**Comment by iSazonov** [comment]
Reopen to restart all CIs (no response).

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 [hunk 3]
```diff
@@ -96,7 +98,15 @@ Describe "Validate that get-help works for CurrentUserScope" -Tags @('CI') {
...
+    It "Last ProgressRecord should be Completed" {
+        $j = Start-Job { Get-Help DoesNotExist }
+        $j | Wait-Job
+        $j.ChildJobs[0].Progress[-1].RecordType | Should -Be ([System.Management.Automation.ProgressRecordType]::Completed)
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: It should be in try-finally.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@powercode Pushed an empty commit to execute `Feature` tests. The new tests are categorized as `Feature`.

**Pull request merged**